### PR TITLE
Allow passing a block as the argument to a dsl_attribute

### PR DESCRIPTION
**Pull request by tlunter**
@nahiluhmot

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 language: ruby
 rvm:
-    - 1.9.2
```
Comment (nahiluhmot) [inline .travis.yml]: Not that I mind, but are we now using a feature in `1.9.3` that isn't in `1.9.2`?

File: lib/dockly/util/dsl.rb [hunk 2]
```diff
@@ -14,9 +14,13 @@ def new!(options = {}, &block)
 
     def dsl_attribute(*names)
       names.each do |name|
-        define_method(name) do |val = nil|
-          val.nil? ? instance_variable_get(:"@#{name}")
-                   : instance_variable_set(:"@#{name}", val)
+        define_method(name) do |val = nil, &block|
```
Comment (nahiluhmot) [inline lib/dockly/util/dsl.rb]: You could keep the same logic as above if you added something like this:

```ruby
val ||= block
val.nil? ? instance_variable_get(:"@#{name}")
         : instance_variable_set(:"@#{name}", val)
```

**Comment by nahiluhmot** [comment]
:+1: Logic looks good.

File: lib/dockly/util/dsl.rb (hunk 2 above)
Comment (tlunter) [inline lib/dockly/util/dsl.rb]: This does not work for a value that is false.

File: .travis.yml (hunk 1 above)
Comment (tlunter) [inline .travis.yml]: The parallel gem required by cane won't build in 1.9.2. Not truly breaking, but we also haven't supported 1.9.2 in the downstreams for a while.

File: lib/dockly/util/dsl.rb (hunk 2 above)
Comment (nahiluhmot) [inline lib/dockly/util/dsl.rb]: Ah, stupid truthiness.

image

**Pull request merged**